File: desafio-02/lucascorreaads/javascript/desafio02.js
//Programa que lista todos números primos entre 1 e 10000

//Declarado o array que receberá os números primos verificados e em seguida imprime o array
let primos = [2]
console.log(2)

//Insere os números entre 2 a 10000 a serem verificados na função
for (let i = 2; i < 10000; i++) {
  verificadorPrimo(i)
}
//Função que verifica se os números são primos
function verificadorPrimo(num) {
  //Loop que descarta se o número não for primo e imprime se for
  for (let pos in primos) {
    //Caso o número inserido tenha resto igual a 0 com algum número primo, ele é descartado
    if (num % primos[pos] == 0) {
      break
      //Se o número não for divisível por nenhum número primo, ele é imprimido e colocado no array
    } else if (pos == primos.length - 1) {
        console.log(num)
        primos.push(num)
    }
  }
}

File: desafio-02/alexandrevarga/python/desafio-02.py
# Números primos


def encontrar_primos(limite_inferior, limite_superior):
    lista_primos = []
    for num in range(limite_inferior, limite_superior + 1):
        if num > 1:
            primo = True
            for i in range(2, int(num ** 0.5) + 1):
                if num % i == 0:
                    primo = False
                    break
            if primo:
                lista_primos.append(num)
    return lista_primos

# Encontrar números primos entre 1 e 10000
primos = encontrar_primos(1, 10000)
print(primos)
